server/weaverbird/backends/sql_translator/steps/convert.py

```diff
@@ -35,15 +35,30 @@ def translate_convert(
         f'query.metadata_manager.query_metadata: {query.metadata_manager.retrieve_query_metadata()}\n'
     )
 
+    to_cast = []
+    for col in step.columns:
+        retrieved_col_type = query.metadata_manager.retrieve_query_metadata_column_type_by_name(
+            column_name=col
+        )
+        if retrieved_col_type == 'FLOAT' and step.data_type == 'integer':
+            to_cast.append(f'TRUNCATE({col}) AS {col}')
+        elif retrieved_col_type == 'STR' and step.data_type == 'integer':
+            to_cast.append(f"CAST(SPLIT_PART({col}, '.', 0) AS {step.data_type}) AS {col}")
+        elif (retrieved_col_type == 'TIMESTAMPE_NTZ') and step.data_type == 'integer':
```

Comment:
```suggestion
        elif (retrieved_col_type == 'TIMESTAMP_NTZ') and step.data_type == 'integer':
```

Ya pas un E en trop ? https://docs.snowflake.com/en/sql-reference/data-types-datetime.html#timestamp-ltz-timestamp-ntz-timestamp-tz

Reply:
:rofl: https://github.com/ToucanToco/toucan-connectors/blob/08f74e119fc155f015aa42d14f892eb5fc0d4c5a/toucan_connectors/snowflake_common.py#L24 
du coup je fais une PR de fix connectors